### ISSUE TITLE
Add more search options to swupd

### ISSUE THREAD

**Issue by castulo** [clearlinux/swupd-client#1158]
Swupd should include options to search for the following things:

- look for a bundle name matching the string   (already covered by `swupd search`)
- look for direct-package matching the string
- look for package matching the string
- look for file matching the string  (already covered by `swupd search-file`)

Related to #461
This issue is blocked by clearlinux/mixer-tools#619.
